src/tree.rs

```diff
@@ -158,6 +158,8 @@ impl<'a> Node<'a> {
         }
     }
 
+    pub fn is_text(&self) -> bool { matches!(self.data, NodeData::Text(_))}
```

Comment:
Is this really useful, compared to `node.as_text().is_some()`? It also encourages redundant tests (see comment on the new iterator.)